libnavigation-core/src/test/java/com/mapbox/navigation/core/internal/HistoryRecordingStateHandlerTest.kt

```diff
@@ -0,0 +1,753 @@
+package com.mapbox.navigation.core.internal
+
+import com.mapbox.navigation.core.HistoryRecordingStateHandler
+import com.mapbox.navigation.core.trip.session.NavigationSessionState
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import io.mockk.Matcher
+import io.mockk.clearMocks
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Before
+import org.junit.Test
+
+class HistoryRecordingStateHandlerTest {
+
+    private val observer = mockk<HistoryRecordingStateChangeObserver>(relaxUnitFun = true)
+    private lateinit var stateHandler: HistoryRecordingStateHandler
+
+    @Before
+    fun setUp() {
+        stateHandler = HistoryRecordingStateHandler(NavigationSessionState.Idle)
+        stateHandler.registerStateChangeObserver(observer)
+    }
+
+    @Test
+    fun observersNotification() {
+        val observer1 = mockk<HistoryRecordingStateChangeObserver>(relaxUnitFun = true)
+        val observer2 = mockk<HistoryRecordingStateChangeObserver>(relaxUnitFun = true)
+        val observer3 = mockk<HistoryRecordingStateChangeObserver>(relaxUnitFun = true)
+        stateHandler = HistoryRecordingStateHandler(NavigationSessionState.Idle)
+
+        // no observers
+        stateHandler.onSessionStateChanged(TripSessionState.STARTED)
+
+        verifyNoInteractions(observer1, observer2, observer3)
+
+        // 1 observer
+        stateHandler.registerStateChangeObserver(observer1)
+        stateHandler.setRoutes(listOf(mockk()))
+
+        verify(exactly = 1) {
+            observer1.onShouldStopRecording(any<NavigationSessionState.FreeDrive>())
+            observer1.onShouldStartRecording(any<NavigationSessionState.ActiveGuidance>())
```

Comment:
`any` doesn't work that way, it matches anything regardless of the specified type. 
```suggestion
            observer1.onShouldStopRecording(ofType<NavigationSessionState.FreeDrive>())
            observer1.onShouldStartRecording(ofType<NavigationSessionState.ActiveGuidance>())
```
The same below.